packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,72 @@
+function get_language(languages_codes) {
+	const parser = Npm.require('accept-language-parser');
+	const languages = Npm.require('languages');
+
+	const languages_parsed = parser.parse(languages_codes);
+	const priority_language = languages_parsed[0].code;
+
+	const priority_language_fullname = languages.getLanguageInfo(priority_language);
+	return priority_language_fullname;
+}
+
+
+function add_or_update_user(channel, plugin_name) {
+	Meteor.users.update(Meteor.userId(), {
+		$push: {
+			automatic_channels: {'name': channel, 'plugin': plugin_name, 'blacklisted': 'false'}
+		}
+	});
+}
+
+function remove_user_from_automatic_channel() {
+	const user_info = RocketChat.models.Users.findOneById(Meteor.userId());
+	// get the name of channels to which user is added
+	const to_be_removed = user_info.automatic_channels.filter(function(obj) {
+		return obj.blacklisted ==='false';
+	});
+	if (to_be_removed.length!==0) {
+		Meteor.users.update({
+			_id: Meteor.userId()
+		}, {
+			$pull: { automatic_channels: { 'name': to_be_removed[0].name } }
```

Comment:
If I do ```blacklisted: false``` , then anyhow the check would still be required here 
``` const room =  RocketChat.models.Rooms.findOneByIdOrName(to_be_removed[0].name);
	RocketChat.removeUserFromRoom(room._id, Meteor.user());
```
to remove the user from the room. It wil be requied to check whether  ```to_be_removed``` is empty or not.